src/backend/distributed/planner/insert_select_planner.c

```diff
@@ -384,7 +384,9 @@ CreateInsertSelectIntoLocalTablePlan(uint64 planId, Query *insertSelectQuery,
 	/* get the SELECT query (may have changed after PrepareInsertSelectForCitusPlanner) */
 	Query *selectQuery = selectRte->subquery;
 
-	DistributedPlan *distPlan = CreateDistributedPlan(planId, selectQuery,
+	List *previousSubPlanList = NIL;
+	DistributedPlan *distPlan = CreateDistributedPlan(planId, previousSubPlanList,
```

Comment:
do we need the whole list or just a boolean?

Reply:
I think `allowRecursivePlanning` sounds much easier to follow / document